File: app/src/main/kotlin/fm/dex/model/entity/Channel.java
package fm.dex.model.entity;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Generated;
import org.greenrobot.greendao.annotation.NotNull;

@Entity
public class Channel {

    @NotNull
    private String title;

    @NotNull
    private String description;

    @NotNull
    private String link;

    @NotNull
    private String rss;

    @NotNull
    private String thumbnail;

    @NotNull
    private String keywords;

    public String getKeywords() {
        return this.keywords;
    }

    public void setKeywords(String keywords) {
        this.keywords = keywords;
    }

    public String getThumbnail() {
        return this.thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    public String getRss() {
        return this.rss;
    }

    public void setRss(String rss) {
        this.rss = rss;
    }

    public String getLink() {
        return this.link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Generated(hash = 1039619931)
    public Channel(@NotNull String title, @NotNull String description,
            @NotNull String link, @NotNull String rss, @NotNull String thumbnail,
            @NotNull String keywords) {
        this.title = title;
        this.description = description;
        this.link = link;
        this.rss = rss;
        this.thumbnail = thumbnail;
        this.keywords = keywords;
    }

    @Generated(hash = 459652974)
    public Channel() {
    }
}

File: app/src/main/java/fm/dex/model/entity/Channel.kt
package fm.dex.model.entity

import org.greenrobot.greendao.annotation.Entity
import org.greenrobot.greendao.annotation.Generated
import org.greenrobot.greendao.annotation.NotNull

@Entity
class Channel {

    @NotNull
    var title: String? = null

    @NotNull
    var description: String? = null

    @NotNull
    var link: String? = null

    @NotNull
    var rss: String? = null

    @NotNull
    var thumbnail: String? = null

    @NotNull
    var keywords: String? = null

    @Generated(hash = 1039619931)
    constructor(@NotNull title: String, @NotNull description: String,
                @NotNull link: String, @NotNull rss: String, @NotNull thumbnail: String,
                @NotNull keywords: String) {
        this.title = title
        this.description = description
        this.link = link
        this.rss = rss
        this.thumbnail = thumbnail
        this.keywords = keywords
    }

    @Generated(hash = 459652974)
    constructor() {
    }
}
